unic/ucd/name/src/name.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+use std::fmt;
+
+use unic_utils::CharDataTable;
+
+
+#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct Name {
+    pieces: &'static [&'static str],
+}
+
+impl Name {
+    pub fn of(ch: char) -> Option<Name> {
+        data::NAMES.find(ch).map(|&pieces| Name { pieces })
```

Comment:
Wondering, since we have this `map()` call in `Age` and now here, does it come with a runtime cost? I'm guessing it should be reduced well in `release` builds that it doesn't matter, but I'm not sure if we need to actually test it before using it for the char-props that are more critical, like normal and bidi components. What do you think?